integration/setup/setup.go

```diff
@@ -48,13 +84,44 @@ func Setup(m *testing.M, config Config) {
 	}
 
 	if !env.KeepResources() {
-		if config.UseDefaultTenant {
-			wait := true
-			err := EnsureTenantClusterDeleted(ctx, env.ClusterID(), config, wait)
+		g, ctx := errgroup.WithContext(ctx)
+
+		g.Go(func() error {
+			o := func() error {
+				err = ensureBastionHostDeleted(ctx, env.ClusterID(), config)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+
+				return nil
+			}
+			b := backoff.NewMaxRetries(10, 1*time.Minute)
+			n := backoff.NewNotifier(config.Logger, ctx)
+
+			err := backoff.RetryNotify(o, b, n)
 			if err != nil {
-				config.Logger.LogCtx(ctx, "level", "error", "message", "failed to delete tenant cluster", "stack", fmt.Sprintf("%#v", err))
-				v = 1
+				return microerror.Mask(err)
+			}
+
+			return nil
+		})
+
+		g.Go(func() error {
+			if config.UseDefaultTenant {
+				wait := true
+				err := EnsureTenantClusterDeleted(ctx, env.ClusterID(), config, wait)
+				if err != nil {
+					return microerror.Mask(err)
+				}
 			}
+
+			return nil
+		})
+
+		err := g.Wait()
+		if err != nil {
+			config.Logger.LogCtx(ctx, "level", "error", "message", err.Error())
+			v = 1
```

Comment:
It feels like a lot of boilerplate code to have this whole chunk of code for deletion repeated before and after.
It would be nice to have this wrapped in a function.

Reply:
That is true and as with basically most of the e2e code, we have to constantly improve the situation. My means of improvements here was now not code structure but test functionality in terms of debugging. When you have a cool idea we can maybe chat a little bit or you suggest a PR for further improvements. I also got a couple of ideas. Here I just want to scope it down a bit for now.

Reply:
Good that we are aware of this. And let's iterate further then.